### PR TITLE
[docs] Document links maybe added after Activity creation

### LINKED ISSUE #5273: Add links after activity (span) creation
# Feature Request

A recent [experimental addition to the OTel specification](https://github.com/open-telemetry/opentelemetry-specification/pull/3678) introduces a new required `AddLink` operation on spans, which allows adding span links after a span was created.

This addition is expected to be declared stable soon (@carlosalberto is driving this), there already exist prototypes for [C++](https://github.com/open-telemetry/opentelemetry-cpp/pull/2380), [Java](https://github.com/open-telemetry/opentelemetry-java/pull/6084), and [Python](https://github.com/open-telemetry/opentelemetry-python/pull/3618). Messaging semantic conventions, which aim at stabilization this year, require this feature.

It should be decided how to make this feature available for .NET users.

**Is your feature request related to a problem?**

As soon as this feature is declared stable, OTel .NET will need to implement it in order to conform to the specification.

**Describe the solution you'd like:**

Ideally, there would be an `Activity.AddLink` method, similar to [`Activity.AddEvent`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.addevent?view=net-8.0) or [`Activity.AddTag`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.addtag?view=net-8.0).

**Describe alternatives you've considered.**

A workaround would be to create a dedicated child span for each link to be added after a span was created. This is cumbersome, as it will drastically increase the number of spans in some cases and obfuscate resulting traces.

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #5273 

Apart from API Doc, a simple test also added!

### CODE REVIEW

File: src/OpenTelemetry.Api/README.md [hunk 1]
```diff
@@ -335,8 +335,10 @@ chose not to sample this activity.
    Apart from the parent-child relation, activities can be linked using
    `ActivityLinks` which represent the OpenTelemetry
    [Links](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#links-between-spans).
-   The linked activities must be provided during the creation time, as shown
-   below.
+   The linked activities are recommended to be provided during the creation
+   time,though links may be added after activity creation time.
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/README.md:339]:
```suggestion
   time, though links may be added after activity creation as well.
```

File: test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs [hunk 2]
```diff
@@ -1303,6 +1303,31 @@ public void BuilderTypeDoesNotChangeTest()
...
+                .AddSource(source.Name)
+                .Build();
+
+        var link1 = new ActivityLink(new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.Recorded));
+        var link2 = new ActivityLink(new ActivityContext(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom(), ActivityTraceFlags.Recorded));
+
+        using (var activity = source.StartActivity("root"))
+        {
+            activity?.AddLink(link1);
+            activity?.AddLink(link1);
```
Comment (CodeBlanch) [inline test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs:1323]:
```suggestion
            activity?.AddLink(link1);
            activity?.AddLink(link2);
```

File: src/OpenTelemetry.Api/README.md [hunk 3]
```diff
@@ -359,7 +361,14 @@ chose not to sample this activity.
...
         default(ActivityContext),
         initialTags,
-        activityLinks);
+        activityLinks); // links provided at creation time.
+
+    // One may add links after activity is created too.
+    var linkedContext3 = new ActivityContext(
+    ActivityTraceId.CreateFromString("01260a70a81e1fa3ad5a8acfeaa0f711"),
+    ActivitySpanId.CreateFromString("34739aa9e2239da1"),
+    ActivityTraceFlags.None);
```
Comment (CodeBlanch) [inline src/OpenTelemetry.Api/README.md:370]:
```suggestion
    var linkedContext3 = new ActivityContext(
        ActivityTraceId.CreateFromString("01260a70a81e1fa3ad5a8acfeaa0f711"),
        ActivitySpanId.CreateFromString("34739aa9e2239da1"),
        ActivityTraceFlags.None);
```

File: src/OpenTelemetry.Api/README.md [hunk 4]
```diff
@@ -335,8 +335,10 @@ chose not to sample this activity.
    Apart from the parent-child relation, activities can be linked using
    `ActivityLinks` which represent the OpenTelemetry
    [Links](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#links-between-spans).
-   The linked activities must be provided during the creation time, as shown
-   below.
+   The linked activities are recommended to be provided during the creation
+   time, though links may be added after activity creation as well.
+   Only links provided during the creation time are accessible to samplers to
+   make sampling decision.
```
[the remark this answers is not in the document]
Comment (reyang) [reply src/OpenTelemetry.Api/README.md:341]: +1, consider changing "as of System.Diagnostics.DiagnosticSource 9.0.0" to "starting from OpenTelemetry 1.10.0"?

**Review by reyang — APPROVED** [review]
LGTM although my approval probably won't count.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Pull request merged**